admin/includes/configuration_checks.php

```diff
@@ -0,0 +1,28 @@
+<?php
+/**
+ * @package admin
+ * @copyright Copyright 2015 Zen Cart Development Team
+ * @copyright Portions Copyright 2003 osCommerce
+ * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
+ * @version 
+ */
+
+  // Function used for configuration checks only
+  function check_configuration($variable, $check_string) { 
+     global $messageStack; 
+     $data = json_decode($check_string, true); 
+     if (!empty($data['error']) && defined($data['error'])) { 
+        $error_msg = constant($data['error']); 
+     } else {
+        $error_msg = 'Validation error'; 
+     }
+     $id = $data['id']; 
+     $options = $data['options']; 
```

Comment:
Can we do a little more input validation here? 

Currently you're expecting `$id` to be an integer, but people who try to extend this further are going to be a bit confused by having an integer value. I might suggest testing whether `$data['id']` is an integer. If so, carry on. If not, test whether the passed string is defined as a constant (like you do for `$data['error']`) and use that. If not defined and not integer, may as well `return` because filter_var is going to do nothing useful if an invalid `$id` is passed to it. 

And then what about perhaps doing some very basic validation on `$data['options']`? Admittedly that's quite a lot more involved, and I'm not sure it's worth a lot of effort beyond testing whether it contains an `options` element which has an array in it.  Beyond that it's probably best to let error-trapping or exception-handling deal with it.